### PR TITLE
MINOR: Appending value to LIST config should not generate empty string with …

### PR DESCRIPTION
**Pull request by chia7712**
We noticed this issue when appending value to `leader.replication.throttled.replicas`. The default value is empty string, so it produces `,0:0` rather than `0:0`. For another, parsing throttled replicas get failed when there is an empty string (see [parseThrottledPartitions](https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/ConfigHandler.scala#L91))

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/server/ConfigAdminManager.scala [hunk 1]
```diff
@@ -496,9 +496,9 @@ object ConfigAdminManager {
           if (!listType(alterConfigOp.configEntry.name, configKeys))
             throw new InvalidConfigurationException(s"Config value append is not allowed for config key: ${alterConfigOp.configEntry.name}")
           val oldValueList = Option(configProps.getProperty(alterConfigOp.configEntry.name))
-            .orElse(Option(ConfigDef.convertToString(configKeys(configPropName).defaultValue, ConfigDef.Type.LIST)))
-            .getOrElse("")
-            .split(",").toList
+            .orElse(Option(ConfigDef.convertToString(configKeys(configPropName).defaultValue, ConfigDef.Type.LIST)).filter(s => s.nonEmpty))
```
Comment (showuon) [inline core/src/main/scala/kafka/server/ConfigAdminManager.scala]: nit: `filter` can be in next line

File: core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala [hunk 2]
```diff
@@ -2480,6 +2480,27 @@ class PlaintextAdminIntegrationTest extends BaseAdminIntegrationTest {
     }
   }
 
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testAppendConfigToEmptyDefaultValue(ignored: String): Unit = {
+    client = Admin.create(createConfig)
+    val topic = "testAppendConfigToEmptyDefaultValue"
+    createTopic(topic)
```
Comment (showuon) [inline core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala]: Could we also add a test for the case that topic config already contains a `LeaderReplicationThrottledReplicasProp` value, to make sure the result is what we expected? Ex: 

```java
    val p = new Properties()
    p.setProperty(LogConfig.LeaderReplicationThrottledReplicasProp, "1:1")
    createTopic(topic, topicConfig = p)
...
    assertEquals("1:1, 0:0", config.value())
```

**Review by showuon — APPROVED** [review]
LGTM! Nice catch! Left some minor comments.

**Comment by chia7712** [comment]
@showuon thanks for all comments. I have addressed them. PTAL

**Review by showuon — APPROVED** [review]
LGTM. Thanks for the fix.

File: core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala [hunk 3]
```diff
@@ -2480,6 +2480,38 @@ class PlaintextAdminIntegrationTest extends BaseAdminIntegrationTest {
...
 
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testAppendConfigToEmptyDefaultValue(ignored: String): Unit = {
+    testAppendConfig(new Properties(), "0:0", "0:0")
+  }
+
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testAppendConfigToExistentValue(ignored: String): Unit = {
```
Comment (dengziming) [inline core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala:2491]: This test case will succeed even without this change, is it only used to improve test coverage?

Comment (chia7712) [reply core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala:2491]: yep

**Comment by chia7712** [comment]
I will merge it tomorrow if no objection.

**Review by dengziming — APPROVED** [review]
LGTM.

**Pull request merged**